# Medidas Separatrizes: Quartis, Decis e Percentis

In [1]:
import pandas as pd
import numpy as np

In [4]:
dados = pd.read_excel("../Arquivos/dados.xlsx")
dados.head(3)

UF  Sexo  Idade  Cor  Anos de Estudo  Renda      Altura
0  11     0     23    8              12    800  1603807616
1  11     1     23    2              12   1150  1739789827
2  11     1     35    8              15    880  1760443822

### Quantile: Divida a variavel em x itens
##### 0.5 é mesmo que dividir em 2, ou seja, MEDIANA

In [10]:
dados.Renda.quantile(0.5)

1200.0

### Para dividir em quartis, são 3 divisoes, ou seja n - 1

### Vamos dividir em quartis, informaremos as 3 medidias de separação

##### Dessa forma, vai dividir em 0.25, 0.50 e 0.75, quatro medidas IGUAIS

In [11]:
dados.Renda.quantile([0.25, 0.5, 0.75])

0.25     788.0
0.50    1200.0
0.75    2000.0
Name: Renda, dtype: float64

### Para dividir em 10 e em 100, fazer do mesmo modo

##### Faremos um for para gerar os 10 numeros pretendidos:

In [15]:
# Para gerar uma lista de de 0.1 a 0.9, faremos:

[i/10 for i in range(1,10)]

[0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]

In [21]:
# Para gerar o numero que queremos, podemos gerar ja tudo junto:

dados.Renda.quantile([i/10 for i in range(1,10)])

0.1     350.0
0.2     788.0
0.3     800.0
0.4    1000.0
0.5    1200.0
0.6    1500.0
0.7    1900.0
0.8    2500.0
0.9    4000.0
Name: Renda, dtype: float64

###  Significa, que 10% dos valores estao abaixo de 350 e 10% dos valores estao acima de 4000, por exemplo

##### Para gerar percentis, faremos de forma analoga:


In [26]:
dados.Renda.quantile([x/100 for x in range(1,100)])

0.01        0.0
0.02        0.0
0.03        0.0
0.04       50.0
0.05      100.0
         ...   
0.95     6000.0
0.96     7000.0
0.97     8000.0
0.98    10000.0
0.99    15000.0
Name: Renda, Length: 99, dtype: float64